src/DynamoRevit/Models/RevitDynamoModel.cs

```diff
@@ -809,44 +807,37 @@ private void TransactionManager_FailuresRaised(FailuresAccessor failuresAccessor
             }
         }
 
-        private void RevitServicesUpdater_ElementsDeleted(
-            Document document, IEnumerable<ElementId> deleted)
+        void RevitServicesUpdater_ElementsUpdated(object sender, ElementUpdateEventArgs e)
         {
-            if (!deleted.Any())
+            //Element addition is not handled by this class.
+            if (e.Operation == ElementUpdateEventArgs.UpdateType.Added)
                 return;
 
+            if (!e.Elements.Any())
+                return;
+
+            bool dynamoTransaction = e.Transactions.Contains(TransactionWrapper.TransactionName);
+
             var nodes = ElementBinder.GetNodesFromElementIds(
-                deleted,
+                e.Elements,
                 CurrentWorkspace,
                 EngineController);
             foreach (var node in nodes)
             {
+                //Don't re-execute the element construction node, if 
+                //this element is modified due to a Dynamo transaction.
+                if (dynamoTransaction && IsConstructorNode(node)) continue;
```

Comment:
Yes because SetRotation caches the rotation value and before re-executing it checks against the input. If the input and the applied rotation is same then it won't update the element.